Auto release functionality

malikovmm opened the pull request:
To release, change the version in plugin.yml and push the changes
When released, all the words "\_\_VERSION__" in all files except .md will be replaced with the release version from plugin.yaml

malikovmm committed:
replace getCurrentTime with now

malikovmm committed:
separate tables by accordions, make table sortable, ui changes

malikovmm committed:
add filter bar, remove accordion, ui changes

malikovmm committed:
change filtration method

malikovmm committed:
fix package name overflow

malikovmm committed:
build: add gulp to project

move js and css code to separated files, add gh actons

malikovmm committed:
build: change the way of version creating

when release all __VERSION__ strings replaces by version from plugin.yaml

minor changes

afdesk commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,70 @@
+name: 'Build'
+
+on: [push]
```
we don't need to run this action for all builds. 
only if `plugin.yaml` was pushed.

afdesk commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,70 @@
+
+      - name: Display read-yaml output
+        run: echo "${{ steps.yaml-data.outputs.data }}"
+      - uses: mukunku/tag-exists-action@v1.2.0
```
do we really need this check?
if we broke `tags` manually we have to fix it manually too, isn't it?
or do i miss any usecases?

afdesk commented on scan2html:
```diff
@@ -3,7 +3,7 @@
 function help() {
   cat << EOS >&2
-scan2html v0.0.1
+scan2html v__VERSION__
```
no-no, we shouldn't replace it this way

afdesk commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,70 @@
+        uses: jacobtomlinson/gha-find-replace@v3
+        if: ${{ steps.checkTag.outputs.exists == 'false' }}
+        with:
+          find: "__VERSION__"
```
no-no, we shouldn't look for this `__VERSION__`

afdesk commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,70 @@
+        run: mv dist/html.tpl .
+      - name: Compress
+        if: ${{ steps.checkTag.outputs.exists == 'false' }}
+        uses: a7ul/tar-action@v1.1.0
```
do we really need such specific action? 
Doesn't `zip` command work?

afdesk commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,70 @@
+            scan2html
+          outPath: scan2html.tar.gz
+      - name: Release
+        uses: softprops/action-gh-release@v1
```
can we use the default action for it?

afdesk replied:
do we really need it? 
I don't remember such action before...

afdesk replied:
[v0.1.15](https://github.com/softprops/action-gh-release/releases/tag/v0.1.15)

malikovmm committed:
build: gh actions refactor

reduce third party actions usage, change the way on version parsing

minor changes

malikovmm committed:
build: remove  a7ul/tar-action

replace a7ul/tar-action with standart tar command

minor changes

afdesk commented on .github/workflows/build.yml:
```diff
@@ -11,60 +14,39 @@ jobs:
+      - name: Display next-version output
+        run: echo "${{ steps.next-version.outputs.next_version }} ${{ steps.next-version.outputs.next_version != '' }}"
+      - name: Create tag v${{steps.next-version.outputs.next_version}}
+        if: ${{ steps.next-version.outputs.next_version != '' }}
```
let's remove these `if` statements. i figured out a better way:

```yaml
jobs:
  check:
    outputs:
      status: ${{ steps.early.conclusion }}
    steps:
      - id: early
        name: Early exit
        run: # exit_with_success
  work:
    needs: check
    if: needs.check.outputs.status == 'success'
    steps:
      - run: foo
      - run: ...
```

afdesk commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,52 @@
+        run: echo "${{ steps.next-version.outputs.next_version }} ${{ steps.next-version.outputs.next_version != '' }}"
+      - name: Create tag v${{steps.next-version.outputs.next_version}}
+        if: ${{ steps.next-version.outputs.next_version != '' }}
+        uses: rickstaa/action-create-tag@v1
```
I took a look at this action. 
ok, let's use it.
but we have to set a specific version: v1.5.4, right?

malikovmm committed:
build: gh actions refactor

reduce number of third-party actions

separate steps on jobs

afdesk commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,69 @@
+      - 'plugin.yaml'
+
+jobs:
+  build:
```
replace build

malikovmm committed:
build: gh action refactor

remove the build if the version was not changed

define softprops/action-gh-release version
